beetsplug/embedart.py

```diff
@@ -86,12 +86,23 @@ def embed_func(lib, opts, args):
                                     help='extract an image from file metadata')
         extract_cmd.parser.add_option('-o', dest='outpath',
                                       help='image output file')
+        extract_cmd.parser.add_option('-a', dest='albums', action='store_true',
+                                      help='extract the art of all matching '
+                                           'albums')
 
         def extract_func(lib, opts, args):
-            outpath = normpath(opts.outpath or config['art_filename'].get())
-            for item in lib.items(decargs(args)):
-                if self.extract(outpath, item):
+            outpath = opts.outpath or config['art_filename'].get()
+            if opts.albums:
+                if opts.outpath and '/' in opts.outpath.replace('\\', '/'):
```

Comment:
I was trying `os.sep` before but it won't work because of the magic of `normpath`... Under windows, `os.sep` is set to `\`. Now we call the command with a path like `D:/cover.jpg` (notice the slash). Searching for `os.sep` will give us no result, hence it seems not to be a valid path, so it must be a filename. Passing this to `normpath` will give us `D:\cover.jpg` (the false slash has been corrected). Now it is a valid path.

We can't execute `normepath` before the test because it will create an absolute path nevertheless, so the test will always match (and cancel the command).

I don't know if there is a nicer way to determine if a string is a path rather than searching for `/` and `\`.